tests/functional/tokens/test_generate.py

```diff
@@ -0,0 +1,231 @@
+    "'scopes' in 'payload'",
+    target_fixture="response",
+)
+def the_admin_sends_request_with_invalid_scopes(
+    http_request, headers, payload
+):
+    response = http_request(
+        method="POST", url="/api/v1/token/new", headers=headers, json=payload
+    )
+    return response
+
+
+@then("the admin should get status code '422'")
+def the_admin_should_get_status_code_422_scopes(response):
+    assert response.status_code == 422, response.text
+
+
+@then("the admin should get not allowed scope as a given context detail")
+def the_admin_should_get_scope_in_given_context_detail(response):
+    not_allowed_scopes = ["write:token", "write:bootstrap"]
+    assert response.json()["detail"][0]["ctx"]["given"] in not_allowed_scopes
+
+
+@scenario(
+    "../../features/tokens/generate.feature",
+    "Admin is Unauthorized to generate using HTTP API with invalid token",
+)
+def test_admin_is_unauthorized_to_generate_invalid_token():
+    """Admin is Unauthorized to generate using HTTP API with invalid token."""
+
+
+@when(
+    "the admin sends a POST request to '/api/v1/token/new' with invalid "
+    "'access_token' in the headers",
+    target_fixture="response",
+)
+def the_admin_sends_a_invalid_access_token_in_headers(
+    http_request, headers, payload
+):
+    response = http_request(
+        method="POST",
+        url="/api/v1/token/new",
+        headers=headers,
+        json=payload,
+    )
+    return response
+
+
+@then("the admin should get status code '401'")
+def the_admin_should_get_status_code_401(response):
+    assert response.status_code == 401, response.text
+
+
+@then("the admin should get 'Failed to validate token' in body")
+def the_admin_should_get_failed_to_validate_token(response):
+    assert response.json()["detail"]["error"] == "Failed to validate token"
+
+
+@scenario(
+    "../../features/tokens/generate.feature",
+    "Admin uses RSTUF Command Line Interface to generate Token",
+)
+def test_admin_uses_rstuf_command_line_interface_to_generate_token():
+    """Admin uses RSTUF Command Line Interface to generate Token."""
+
+
+@given("the admin has repository-service-tuf (rstuf) installed")
+def the_admin_has_repositoryservicetuf_rstuf_installed(rstuf_cli):
+    """the admin has repository-service-tuf (rstuf) installed."""
+    rc, stdout = rstuf_cli("-h")
+    assert rc == 0, stdout
+
+
+@given("the admin is logged in using RSTUF Command Line Interface")
+def the_admin_is_logged_in_using_rstuf_command_line_interface(rstuf_cli):
+    """the admin is logged in using RSTUF Command Line Interface."""
+    rc, stdout = rstuf_cli(
+        "admin login -f -s http://localhost -u admin -p secret -e 1"
+    )
+
+    assert rc == 0, stdout
+    assert "Login successful" in stdout
+
+
+@given(
+    parse(
+        "the admin types 'rstuf -c rstuf.ini admin token generate "
+        "-s {scopes_params} -e {expires_params}"
+    ),
+    target_fixture="cli_parameters",
```

Comment:
I guess not but is the value of `cli_parameters` affected by how we use quotes (related to my previous comment)?